docs/designs/link-output.md

```diff
@@ -0,0 +1,97 @@
+# Link And Bookmarks Output
+
+Docfx generate `links` and `bookmarks` information of each files into one single file `links.json` for future data utilization of downstream service of docfx.
+
+## Links
+
+`links` includes all the links which has been processed by docfx:
+
+- Relative link in source content will be processed by docfx and convert it to absolute link if it's valid, otherwise keep it untouched.
+
+- Absolute link in source content will NOT processed by docfx
+
+| Links in source content | Examples              | Output in `links.json` |
+| ----------------------- | --------------------- | ---------------------- |
+| Relative Link           | ./a.md (valid)        | /basepath/article/a    |
+|                         | a/b.md (invalid)      | a/b.md                 |
```

Comment:
Is `a/b.md` input or output? I think even an invalid input `a/b.md` will be `a/b` in output, then the output in `links.json` is `a/b`.

Reply:
it's input.
In v3, we don't touch any links which are invalid.